lib/block_view_test.go

```diff
@@ -14690,3 +15303,1854 @@ func TestUpdateProfileChangeBack(t *testing.T) {
+	}
+
+	// Have m1, m2, and m3 bid on <post #2, #1> (which has an unlockable).
+	{
+		bidEntries := DBGetNFTBidEntries(db, post2Hash, 1)
+		require.Equal(0, len(bidEntries))
+
+		_createNFTBidWithTestMeta(
+			testMeta,
+			10, /*FeeRateNanosPerKB*/
+			m1Pub,
+			m1Priv,
+			post2Hash,
+			1, /*SerialNumber*/
+			5, /*BidAmountNanos*/
+		)
+
+		bidEntries = DBGetNFTBidEntries(db, post2Hash, 1)
+		require.Equal(1, len(bidEntries))
+
+		_createNFTBidWithTestMeta(
+			testMeta,
+			10, /*FeeRateNanosPerKB*/
+			m2Pub,
+			m2Priv,
+			post2Hash,
+			1,  /*SerialNumber*/
+			10, /*BidAmountNanos*/
+		)
+
+		bidEntries = DBGetNFTBidEntries(db, post2Hash, 1)
+		require.Equal(2, len(bidEntries))
+
+		// m1 updates their bid to outbid m2.
+		_createNFTBidWithTestMeta(
+			testMeta,
+			10, /*FeeRateNanosPerKB*/
+			m1Pub,
+			m1Priv,
+			post2Hash,
+			1,  /*SerialNumber*/
+			11, /*BidAmountNanos*/
+		)
+
+		// The number of bid entries should not change since this is just an update.
+		bidEntries = DBGetNFTBidEntries(db, post2Hash, 1)
+		require.Equal(2, len(bidEntries))
+
+		_createNFTBidWithTestMeta(
+			testMeta,
+			10, /*FeeRateNanosPerKB*/
+			m3Pub,
+			m3Priv,
+			post2Hash,
+			1,  /*SerialNumber*/
+			12, /*BidAmountNanos*/
+		)
+
+		bidEntries = DBGetNFTBidEntries(db, post2Hash, 1)
+		require.Equal(3, len(bidEntries))
+
+		// m1 updates their bid to outbid m3.
+		_createNFTBidWithTestMeta(
+			testMeta,
+			10, /*FeeRateNanosPerKB*/
+			m1Pub,
+			m1Priv,
+			post2Hash,
+			1,  /*SerialNumber*/
+			13, /*BidAmountNanos*/
+		)
+
+		bidEntries = DBGetNFTBidEntries(db, post2Hash, 1)
+		require.Equal(3, len(bidEntries))
+	}
+
+	// Error case: can't accept a bid for an unlockable NFT, without providing the unlockable.
+	{
+		_, _, _, err = _acceptNFTBid(
+			t, chain, db, params, 10,
+			m0Pub,
+			m0Priv,
+			post2Hash,
+			1, /*SerialNumber*/
+			m3Pub,
+			12, /*BidAmountNanos*/
+			"", /*UnencryptedUnlockableText*/
+		)
+		require.Error(err)
+		require.Contains(err.Error(), RuleErrorUnlockableNFTMustProvideUnlockableText)
+	}
+
+	{
+		unencryptedUnlockableText := "this is an unlockable string"
+
+		// Accepting the bid with an unlockable string should work.
+		_acceptNFTBidWithTestMeta(
+			testMeta,
+			10, /*FeeRateNanosPerKB*/
+			m0Pub,
+			m0Priv,
+			post2Hash,
+			1,                         /*SerialNumber*/
+			m3Pub,                     /*bidderPkBase58Check*/
+			12,                        /*BidAmountNanos*/
+			unencryptedUnlockableText, /*UnencryptedUnlockableText*/
+		)
+
+		// Check and make sure the unlockable looks gucci.
+		nftEntry := DBGetNFTEntryByPostHashSerialNumber(db, post2Hash, 1)
+		priv, _ := btcec.PrivKeyFromBytes(btcec.S256(), _strToPk(t, m3Priv))
+		decryptedBytes, err := DecryptBytesWithPrivateKey(nftEntry.UnlockableText, priv.ToECDSA())
+		require.NoError(err)
+		require.Equal(unencryptedUnlockableText, string(decryptedBytes))
+		require.Equal(nftEntry.IsForSale, false)
+	}
```

Comment:
should add test to make sure other bids are cancelled when bid is accepted.

Reply:
See "TestNFTBidsAreCanceledAfterAccept."